R/scm_support.R

```diff
@@ -44,38 +43,50 @@ equilibrium_quiet <- function(base=Control()) {
   base
 }
 
+##' Hopefully sensible set of parameters for use with the SCM.  Turns
+##' accuracy down a bunch, makes it noisy, sets up the
+##' hyperparameterisation that we most often use.
+##' @title Sensible, fast (ish) SCM parameters
+##' @author Rich FitzJohn
+##' @param type Name of model (defaults to FF16 but any strategy name is valid).
+##' @export
+scm_base_control <- function(type="FF16", env=environment_type(type)) {
+  ctrl <- equilibrium_verbose(fast_control())
+  ctrl$schedule_eps <- 0.005
+  ctrl$equilibrium_eps <- 1e-3
+  return(ctrl)
+}
+
+
+##' This used to be bundled into base control but refactoring to separate
+##' concerns. May disappear in the future.
+scm_base_parameters <- function(type="FF16", env=environment_type(type)) {
+   Parameters(type, env)(patch_area=1.0)
+}
+
+
 ##' Run the SCM, returning the SCM object for interrogation
 ##'
 ##' This is the simplest way of using the SCM, probably.
 ##' @title Run SCM
 ##' @param p Parameters object
+##' @param env Environment object (defaults to FF16_Environment)
+##' @param ctrl Control object
 ##' @param use_ode_times Should ODE times be used?
 ##' @return A \code{SCM} object.
 ##' @author Rich FitzJohn
 ##' @export
-run_scm <- function(p, use_ode_times=FALSE) {
+run_scm <- function(p, env = make_environment(), 
+                    ctrl = scm_base_control(), use_ode_times=FALSE) {
```

Comment:
As with `build_schedule` how will `make_environment` know what type of environment to make? Same for `scm_base_control`. Would it work to change default to something like  `function(p, env = make_environment(p$type), ctrl = scm_base_control(p$type))`?